src/multisketch.rs

```diff
@@ -196,6 +191,67 @@ impl MultiSketch {
 
         self
     }
+    pub fn remove_metadata(
+        &mut self,
+        output_file_name: &str,
+        genome_ids_to_remove: &[String],
+    ) -> std::io::Result<()> {
+        println!("{}", self);
+        let mut new_sketch_metadata: Vec<Sketch> = Vec::with_capacity(self.sketch_metadata.len());
+
+        for sketch in &self.sketch_metadata {
+            if !genome_ids_to_remove.contains(&(*sketch.name()).to_string()) {
+                new_sketch_metadata.push(sketch.clone());
+            }
+        }
+        self.sketch_metadata = new_sketch_metadata;
+        let _ = self.save_metadata(output_file_name);
+        Ok(())
+    }
+
+    pub fn remove_genomes(
+        &mut self,
+        input_prefix: &str,
+        output_file: &str,
+        genome_ids_to_remove: &[String],
+    ) -> std::io::Result<()> {
+        // Check if all genome IDs to remove exist and get their positions
+        let mut positions_to_remove = Vec::new();
+        let mut missing_ids = Vec::new();
+
+        for id in genome_ids_to_remove {
+            println!("{}", id);
+            if let Some(&position) = self.name_map.get(id) {
+                positions_to_remove.push(position);
+            } else {
+                missing_ids.push(id);
```

Comment:
Ideally you'd do this check earlier and more consistently between the metadata delete and sketch delete functions. I think you can leave this as-is, but I've made a suggestion to add this to the metadata delete function too.